src/mpid/ch4/netmod/ucx/ucx_recv.h

```diff
@@ -37,7 +37,14 @@ static inline void MPIDI_UCX_recv_cmpl_cb(void *request, ucs_status_t status,
 
     if (unlikely(status == UCS_ERR_CANCELED)) {
         MPIR_STATUS_SET_CANCEL_BIT(rreq->status, TRUE);
-    } else {
+    }
+    else if (unlikely(status == UCS_ERR_MESSAGE_TRUNCATED)) {
```

Comment:
Can we just move this check inside the `else` block? I'm not sure it's necessary to have this block and duplicate assignments for source and tag.